pkg/frontend/querymiddleware/codec.go

```diff
@@ -438,3 +441,8 @@ func decorateWithParamName(err error, field string) error {
 	}
 	return apierror.Newf(apierror.TypeBadData, errTmpl, field, err)
 }
+
+func mustReadAll(r io.Reader) []byte {
+	body, _ := ioutil.ReadAll(r)
```

Comment:
could this use `bodyBuffer(r)`, which `DecodeResponse()` calls on the non-error case ?